proto/zitadel/user/schema/v1alpha/user_schema.proto

```diff
@@ -0,0 +1,117 @@
+    }
+  ];
+  // Details provide some base information (such as the last change date) of the schema.
+  zitadel.object.v2beta.Details details = 2;
+  // Type is a human readable text describing the schema.
+  string type = 3 [
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      example: "\"employees\""
+    }
+  ];
+  // Current state of the schema.
+  State state = 4 [
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      example: "\"STATE_ACTIVE\""
+    }
+  ];
+  // Revision is a read only version of the schema, each update increases the revision.
+  uint32 revision = 5 [
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      example: "\"2\""
+    }
+  ];
+  // JSON schema representation defining the user.
+  google.protobuf.Struct schema = 6 [
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      example: "{\"$schema\":\"https://example.com/user/employees\",\"type\":\"object\",\"properties\":{\"name\":{\"type\":\"string\",\"required\":true},\"description\":{\"type\":\"string\"}}}"
+    }
+  ];
+  // Defines the possible types of authenticators.
+  // This allows creating different user types like human/machine without usage of actions to validate possible authenticators.
+  // Removal of an authenticator does not remove the authenticator on a user.
+  repeated AuthenticatorType possible_authenticators = 7  [
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      example: "[\"AUTHENTICATOR_TYPE_USERNAME\",\"AUTHENTICATOR_TYPE_PASSWORD\",\"AUTHENTICATOR_TYPE_WEBAUTHN\"]";
+    }
+  ];
+}
+
+enum FieldName {
+  FIELD_NAME_UNSPECIFIED = 0;
+  FIELD_NAME_TYPE = 1;
+  FIELD_NAME_STATE = 2;
+  FIELD_NAME_REVISION = 3;
+  FIELD_NAME_CREATION_DATE = 4;
+}
+
+message SearchQuery {
+  oneof query {
+    option (validate.required) = true;
+
+    // Limit the result to a specific schema type.
+    TypeQuery type_query = 1;
+    // Limit the result to a specific state of the schema.
+    StateQuery state_query = 2;
+  }
+}
+
+message TypeQuery {
+  // Defines which type to query for.
+  string type = 1 [
+    (validate.rules).string = {max_len: 200},
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      max_length: 200,
+      example: "\"employees\"";
+    }
+  ];
+  // Defines which text comparison method used for the type query.
+  zitadel.object.v2beta.TextQueryMethod method = 2 [
+    (validate.rules).enum.defined_only = true
+  ];
+}
+
+message StateQuery {
+  // Defines the state to query for.
+  State state = 1 [
+    (validate.rules).enum.defined_only = true,
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      example: "\"STATE_ACTIVE\""
+    }
+  ];
+}
+
+enum State {
+  STATE_UNSPECIFIED = 0;
+  STATE_ACTIVE = 1;
+  STATE_INACTIVE = 2;
+}
+
+enum AuthenticatorType {
+  AUTHENTICATOR_TYPE_UNSPECIFIED = 0;
+  //(--TODO: i guess we can also disable this?--)
```

Comment:
i guess we can also disable usernames?

Reply:
Disable username in what context? So that they can't be used as authenticator?

Reply:
yes. i mean it would not be necessary for a federated user. but the UX of the login flow might suffer.
since we might need it in other places, i'll keep it

Reply:
True, as we sort of still need something human-readable for the login, maybe we would also need to change the sessions API to authenticators?